.travis.yml

```diff
@@ -5,143 +5,65 @@ python:
-    - "TESTDIR=Java/vertx"
-    - "TESTDIR=Java/wicket"
-    - "TESTDIR=Java/wildfly-ee7"
-    - "TESTDIR=JavaScript/express"
-    - "TESTDIR=JavaScript/hapi"
-    - "TESTDIR=JavaScript/nodejs"
-    - "TESTDIR=JavaScript/ringojs"
-    - "TESTDIR=JavaScript/ringojs-convenient"
-    - "TESTDIR=Lua/lapis"
-    - "TESTDIR=Lua/openresty"
-    - "TESTDIR=Nim/jester"
-    - "TESTDIR=Nim/nawak"
-    - "TESTDIR=Perl/dancer"
-    - "TESTDIR=Perl/kelp"
-    - "TESTDIR=Perl/mojolicious"
-    - "TESTDIR=Perl/plack"
-    - "TESTDIR=Perl/web-simple"
-    - "TESTDIR=PHP/cakephp"
-    - "TESTDIR=PHP/hhvm"
-    - "TESTDIR=PHP/php"
-    - "TESTDIR=PHP/codeigniter"
-    - "TESTDIR=PHP/php-fatfree"
-    - "TESTDIR=PHP/fuel"
-    - "TESTDIR=PHP/kohana"
-    - "TESTDIR=PHP/php-laravel"
-    - "TESTDIR=PHP/lithium"
-    - "TESTDIR=PHP/php-micromvc"
-    - "TESTDIR=PHP/php-phalcon"
-    - "TESTDIR=PHP/php-phalcon-micro"
-    - "TESTDIR=PHP/php-phpixie"
-    - "TESTDIR=PHP/php-pimf"
-    - "TESTDIR=PHP/php-silex"
-    - "TESTDIR=PHP/php-silex-orm"
-    - "TESTDIR=PHP/php-slim"
-    - "TESTDIR=PHP/symfony2"
-    - "TESTDIR=PHP/symfony2-stripped"
-    - "TESTDIR=PHP/php-yaf"
-    - "TESTDIR=PHP/Yii2"
-    - "TESTDIR=PHP/php-zend-framework"
-    - "TESTDIR=PHP/php-zend-framework1"
-    - "TESTDIR=PHP/phreeze"
-    - "TESTDIR=Python/API-Hour"
-    - "TESTDIR=Python/bottle"
-    - "TESTDIR=Python/django"
-    - "TESTDIR=Python/falcon"
-    - "TESTDIR=Python/flask"
-    - "TESTDIR=Python/pyramid"
-    - "TESTDIR=Python/tornado"
-    - "TESTDIR=Python/uwsgi"
-    - "TESTDIR=Python/wsgi"
-    - "TESTDIR=Racket/racket-ws"
-    - "TESTDIR=Ruby/grape"
-    - "TESTDIR=Ruby/ngx_mruby"
-    - "TESTDIR=Ruby/padrino"
-    - "TESTDIR=Ruby/rack"
-    - "TESTDIR=Ruby/rails"
-    - "TESTDIR=Ruby/rails-stripped"
-    - "TESTDIR=Ruby/sinatra"
-    - "TESTDIR=Scala/finagle"
-    - "TESTDIR=Scala/lift-stateless"
-    - "TESTDIR=Scala/plain"
-    - "TESTDIR=Scala/play-activate-mysql"
-    - "TESTDIR=Scala/play2-scala"
-    - "TESTDIR=Scala/scalatra"
-    - "TESTDIR=Scala/scruffy"
-    - "TESTDIR=Scala/spray"
-    - "TESTDIR=Scala/unfiltered"
-    - "TESTDIR=Ur/urweb"
+    - "TESTDIR=C\\\\/"
+    - "TESTDIR=CSharp"
+    # C++: cpoll_cppsp,treefrog,ULib
+    - "TESTDIR=C\\\\+\\\\+\\\\/[ctU]"
+    # C++: wt
+    - "TESTDIR=C\\\\+\\\\+\\\\/wt"
+    - "TESTDIR=Clojure"
+    - "TESTDIR=Dart"
+    - "TESTDIR=Elixir"
+    - "TESTDIR=Erlang"
+    - "TESTDIR=Go"
+    - "TESTDIR=Groovy"
+    - "TESTDIR=Haskell"
+    # Java: activeweb to dropwizard
+    - "TESTDIR=Java\\\\/[a-d]"
+    # Java: gemini to ninja
+    - "TESTDIR=Java\\\\/[g-n]"
+    # Java: play to restexpress
+    - "TESTDIR=Java\\\\/[p-r]"
+    # Java: servlet to tapestry
+    - "TESTDIR=Java\\\\/[st]"
+    # Java: undertow to wildfly
+    - "TESTDIR=Java\\\\/[u-w]"
+    - "TESTDIR=JavaScript"
+    - "TESTDIR=Lua"
+    - "TESTDIR=Nim"
+    - "TESTDIR=Perl"
+    # PHP: cakephp to hhvm
+    - "TESTDIR=PHP\\\\/[c-h]"
```

Comment:
Because language partitioning is impossible here - it overruns our 50 minute time limit with Travis. And yes, it will be hell to maintain. This is why I'm saying in the conversation thread that we absolutely *must* replace our usage of the public travis-ci in some way - this is clearly an untenable `.travis.yml`, but its the only `.travis.yml` that won't consume huge resources on their hardware